docs/source/getting-started/tabular-data/data-prep.ipynb

```diff
@@ -0,0 +1,272 @@
+       "┌──────────────────────────┬───────┐\n",
+       "│ ILOSTAT                  ┆ len   │\n",
+       "│ ---                      ┆ ---   │\n",
+       "│ str                      ┆ u32   │\n",
+       "╞══════════════════════════╪═══════╡\n",
+       "│ Laid Off                 ┆ 75639 │\n",
+       "│ NA                       ┆ 35922 │\n",
+       "│ Working for Pay          ┆ 79844 │\n",
+       "│ Not Working But Employed ┆ 8002  │\n",
+       "└──────────────────────────┴───────┘"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dp_values = context.query().group_by(\"ILOSTAT\").agg(pl.len().dp.noise()).release().collect()\n",
+    "\n",
+    "labor_status_codes = {\n",
+    "    9: 'NA',\n",
+    "    2: 'Not Working But Employed',\n",
+    "    1: 'Working for Pay', \n",
+    "    3: 'Laid Off'\n",
+    "}\n",
+    "\n",
+    "dp_values= dp_values.with_columns(\n",
+    "    pl.col(\"ILOSTAT\").map_elements(lambda x: labor_status_codes.get(x, x))\n",
+    ")\n",
+    "\n",
+    "dp_values"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Although the use of `with_columns` is not directly in a DP query, it does have some important implications that help maintain the DP unit of privacy such as: \n",
+    "\n",
+    "### 1. Number of Rows is Unchanged. \n",
+    "\n",
+    "This is crucial because differential privacy ensures the privacy guarantee is based on the key entity's contributions specified in the compositor. Keeping the row count consistent prevents unintentional changes in the key entity's contributions. \n",
+    "\n",
+    "### 2. Ordering of Rows is Unchanged. \n",
+    "\n",
+    "The ordering of the rows is important because it also ensures consistency in the key entity's contribution. If the order was changed, through a sort for instance, then the users' contribution could also change, diminishing the privacy guarantee. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Limitations of `filter`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Filtering can be thought of as a type of partition. When filtering results in a smaller subset, it can lead to an increase in noise and a decrease in utility.\n",
+    "\n",
+    "Moreover, if you were to filter by a key with public lengths and then group by that key, you effectively introduce two layers of partitioning. These multiple layers of partitioning can significantly diminish the privacy guarantee since the individual partitions are smaller so more noise must be added to each layer to meet the privacy guarantee. When the two layers are combined, the compounded noise further contributes to the decreased utility. \n",
+    "\n",
+    "This is demonstrated by the following example which computes the total count of respondents not working due to a layoff per year. "
```

Comment:
Am I correct in thinking that the intended contrast is between this chart, and the total counts without the filters? I'm not sure that the reader will know what it "should" be, if that's the comparison you intend to make.

I also wonder if the accuracy utilities could be used here, rather than just seeing that the line is subjectively noisier?

Reply:
i clarified the meaning of the graph, not sure if accuracy belongs here since it comes later in the ordering of the pages